### ISSUE TITLE
[google_sign_in] Scroll bug on IOS

### ISSUE THREAD

**Issue by mcfugger** [flutter/flutter#101815]
## Details

Machine: Apple MacBook Pro 14'' M1 2021 Monterey.
Xcode version: 13.3

Steps to reproduce:
Add plugin as a dependency, configure the app with related instructions, run the app on any simulator. You will see any Scrollable widget goes down no matter of applied scroll gesture. Removing the plugin fixes the issue. It ruins developer experience.

Maybe related: https://stackoverflow.com/questions/70654638/flutter-listview-double-bouncing-issue

**Target Platform:**iOS
**Target OS version/browser:** 15.4
**Devices:** iOS Simulator

## Logs

<details>
<summary>Logs</summary>

```
```

[✓] Flutter (Channel stable, 2.10.3, on macOS 12.1 21C52 darwin-arm, locale
    en-TM)
    • Flutter version 2.10.3 at /Users/kerimamansaryyev/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7e9793dee1 (6 weeks ago), 2022-03-02 11:23:12 -0600
    • Engine revision bd539267b4
    • Dart version 2.16.1
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version
    32.1.0-rc1)
    • Android SDK at /Users/kerimamansaryyev/Library/Android/sdk
    • Platform android-32, build-tools 32.1.0-rc1
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)

[✓] VS Code (version 1.65.2)
    • VS Code at /Users/kerimamansaryyev/Downloads/Visual Studio
      Code.app/Contents
    • Flutter extension version 3.38.1

[✓] Connected device (2 available)
    • iPhone 13 Pro Max (mobile) • A03138E6-EB1A-4CDC-AFDE-0D0E381DB92F • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-15-4 (simulator)
    • Chrome (web)               • chrome                               •
      web-javascript • Google Chrome 100.0.4896.75

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

```
```

</details>

**Comment by maheshmnj** [comment]
Hi @mcfugger, Thanks for filing the issue. The issue you are running into I believe is this When you scroll slightly the scroll offset is too high is that correct?
  
https://user-images.githubusercontent.com/31410839/163179118-5e467549-9b28-44a4-a480-5cc4f55d9180.mov

If so, This Looks similar to https://github.com/flutter/flutter/issues/79641, https://github.com/flutter/flutter/issues/93177


<details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key, required this.title}) : super(key: key);
  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Material(
      child: ListView(
        children: [
          for (int i = 0; i < 20; i++)
            ListTile(
              title: Text('Item $i'),
            ),
        ],
      ),
    );
  }
}

```
</details>

**Comment by mcfugger** [comment]
> Hi @mcfugger, Thanks for filing the issue. The issue you are running into I believe is this When you scroll slightly the scroll offset is too high is that correct? Can you please attach the output of `flutter doctor -v`?
> 
>  Screen.Recording.2022-04-13.at.5.53.01.PM.mov 
> 
> If so, This Looks similar to #79641, #93177
> 
> code sample
> flutter doctor -v (mac)

Yes, exactly. I attached flutter doctor -v logs above in Logs tab.

**Comment by mcfugger** [comment]
I found interesting comments here https://github.com/flutter/flutter/issues/93177#issuecomment-967249113. Does the plugin's native sdk support M1 architecture?

**Comment by maheshmnj** [comment]
Looks like it doesn't yet support based on this thread https://developer.apple.com/forums/thread/668320. This issue reproduces on both the stable and the master channel. 

@mcfugger AFAICT from the old issues linked above, this only manifests on a simulator running on Apple silicon. So this shouldn't affect your production apps. Labeling this issue for further investigation from the team

Reproduction details can be found here https://github.com/flutter/flutter/issues/101815#issuecomment-1097994564

cc: @jmagman

**Comment by jmagman** [comment]
> I found interesting comments here [#93177 (comment)](https://github.com/flutter/flutter/issues/93177#issuecomment-967249113). Does the plugin's native sdk support M1 architecture?

Yes, it's exactly the same issue as https://github.com/flutter/flutter/issues/93177, `google_sign_in` also doesn't support arm64 simulators so it's making your whole app drop back to use Rosetta, which has this bug.  Updating the plugin is tracked in https://github.com/flutter/flutter/issues/86436 blocked on https://github.com/google/GoogleSignIn-iOS/issues/23.
